addons/portal_sale/views/account_invoice_views.xml

```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_account_invoice_filter_share" model="ir.ui.view">
+        <field name="name">account.invoice.search.account</field>
+        <field name="model">account.invoice</field>
+        <field name="inherit_id" ref="account.view_account_invoice_filter"/>
+        <field name="arch" type="xml">
+            <xpath expr='//field[@name="user_id"]' position="attributes">
+                <attribute name="domain">[('share','=', False)]</attribute>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="portal_action_invoices" model="ir.actions.act_window">
+        <field name="name">Invoices</field>
+        <field name="res_model">account.invoice</field>
+        <field name="view_mode">tree,form</field>
+        <field name="domain">[('type','in',['out_invoice','out_refund'])]</field>
+        <field name="context">{}</field>
+        <field name="search_view_id" ref="account.view_account_invoice_filter"/>
+        <field name="help">We haven't sent you any invoice.</field>
+    </record>
+
+    <record id="portal_action_invoices_tree_spec" model="ir.actions.act_window.view">
+        <field name="act_window_id" ref="portal_action_invoices"/>
+        <field name="view_id" ref="account.invoice_tree"/>
+        <field name="view_mode">tree</field>
+        <field name="sequence" eval="0"/>
+    </record>
+
+    <record id="portal_action_invoices_form_spec" model="ir.actions.act_window.view">
+        <field name="act_window_id" ref="portal_action_invoices"/>
+        <field name="view_id" ref="account.invoice_form"/>
+        <field name="view_mode">form</field>
+        <field name="sequence" eval="1"/>
+    </record>
+
+    <menuitem name="Invoices" id="portal_invoices" parent="portal.portal_orders"
```

Comment:
Id should be first : 
<menuitem id="portal_invoices" name="Invoices"  parent="portal.portal_orders" ...